src/preprocess.rs

```diff
@@ -150,7 +150,8 @@ mod sys {
                     SpeexPreprocessConst::SPEEX_PREPROCESS_SET_ECHO_STATE,
                     Variant::Echo(val),
                 ) => val.get_ptr() as *mut c_void,
-                (_, Variant::U32(val)) => val as *mut c_void,
+                (_, Variant::U32(val),
+                ) => &(val as f32) as *const f32 as *mut c_void,
```

Comment:
When we get in that branch exactly?

Reply:
When the user inserts any other `SpeexPreprocessConst` with an integer value. 

In the `testdenoise` example we have the following cases: 
```
st.preprocess_ctl(SPEEX_PREPROCESS_SET_DENOISE, 1).unwrap();
st.preprocess_ctl(SPEEX_PREPROCESS_SET_AGC, 0).unwrap();
st.preprocess_ctl(SPEEX_PREPROCESS_SET_AGC_LEVEL, 8000).unwrap();
st.preprocess_ctl(SPEEX_PREPROCESS_SET_DEREVERB, 0).unwrap();
```

Reply:
I see, `&val as *mut c_void` does not work?

Reply:
Get the following error: `casting &u32 as *mut libc::c_void is invalid`

Reply:
right, [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9f77d1f5a41e2162f4675f4c39a10c79) should work.

Reply:
Yeah, it works, thanks!

But is it necessary? What's the problem converting the value into an `f32`?

Reply:
consistency and it _might_ be unsound in certain architectures.

Reply:
Ok, thanks for the explanation!

Done.